cli.js

```diff
@@ -56,94 +57,105 @@ try {
 
 const appInfo = plist.parse(infoPlist);
 const appName = appInfo.CFBundleDisplayName || appInfo.CFBundleName;
-// `const appIconName = appInfo.CFBundleIconFile.replace(/\.icns/, '');
+const appIconName = appInfo.CFBundleIconFile.replace(/\.icns/, '');
 const dmgPath = path.join(destPath, `${appName} ${appInfo.CFBundleShortVersionString}.dmg`);
 
 const ora = new Ora('Creating DMG');
 ora.start();
 
-if (cli.flags.overwrite) {
-	try {
-		fs.unlinkSync(dmgPath);
-	} catch (_) {}
-}
-
-const ee = appdmg({
-	target: dmgPath,
-	basepath: process.cwd(),
-	specification: {
-		title: appName,
-		// Disabled because of #16
-		// icon: path.join(appPath, 'Contents/Resources', `${appIconName}.icns`),
-		//
-		// Use transparent background and `background-color` option when this is fixed:
-		// https://github.com/LinusU/node-appdmg/issues/135
-		background: path.join(__dirname, 'assets/dmg-background.png'),
-		'icon-size': 160,
-		format: 'ULFO',
-		window: {
-			size: {
-				width: 660,
-				height: 400
-			}
-		},
-		contents: [
-			{
-				x: 180,
-				y: 170,
-				type: 'file',
-				path: appPath
-			},
-			{
-				x: 480,
-				y: 170,
-				type: 'link',
-				path: '/Applications'
-			}
-		]
+async function init() {
+	if (cli.flags.overwrite) {
+		try {
+			fs.unlinkSync(dmgPath);
+		} catch (_) {}
 	}
-});
 
-ee.on('progress', info => {
-	if (info.type === 'step-begin') {
-		ora.text = info.title;
-	}
-});
+	ora.text = 'Creating icon';
+	const composedIconPath = await composeIcon(path.join(appPath, 'Contents/Resources', `${appIconName}.icns`));
+
+	const ee = appdmg({
+		target: dmgPath,
+		basepath: process.cwd(),
+		specification: {
+			title: appName,
+			icon: composedIconPath,
+			//
+			// Use transparent background and `background-color` option when this is fixed:
+			// https://github.com/LinusU/node-appdmg/issues/135
+			background: path.join(__dirname, 'assets/dmg-background.png'),
+			'icon-size': 160,
+			format: 'ULFO',
+			window: {
+				size: {
+					width: 660,
+					height: 400
+				}
+			},
+			contents: [
+				{
+					x: 180,
+					y: 170,
+					type: 'file',
+					path: appPath
+				},
+				{
+					x: 480,
+					y: 170,
+					type: 'link',
+					path: '/Applications'
+				}
+			]
+		}
+	});
 
-ee.on('finish', async () => {
-	ora.text = 'Code signing DMG';
-
-	try {
-		let identity;
-		const {stdout} = await execa('security', ['find-identity', '-v', '-p', 'codesigning']);
-		if (stdout.includes('Developer ID Application:')) {
-			identity = 'Developer ID Application';
-		} else if (stdout.includes('Mac Developer:')) {
-			identity = 'Mac Developer';
-		} else {
-			const err = new Error();
-			err.stderr = 'No usable identity found';
-			throw err;
+	ee.on('progress', info => {
+		if (info.type === 'step-begin') {
+			ora.text = info.title;
 		}
+	});
+
+	ee.on('finish', async () => {
+		try {
+			ora.text = 'Replacing DMG icon';
+			await execa('./seticon', [composedIconPath, dmgPath]);
```

Comment:
```suggestion
			await execa(path.join(__dirname, 'seticon'), [composedIconPath, dmgPath]);
```

Also add a comment with the URL to the source of the binary so people know what it is.